recommenders/evaluation/spark_evaluation.py

```diff
@@ -608,12 +618,12 @@ def _get_pairwise_items(self, df):
         )
 
     def _get_cosine_similarity(self, n_partitions=200):
-        if self.item_feature_df is None:
-            # calculate item-item similarity based on item co-occurrence count
-            self._get_cooccurrence_similarity(n_partitions)
-        else:
+        if self.item_sim_measure == "item_feature_vector":
             # calculate item-item similarity based on item feature vectors
             self._get_item_feature_similarity(n_partitions)
+        else:
```

Comment:
Here it is better to check whether it is equal to "item_cooccurrence_count". If it is a third value then you could raise an exception.

Reply:
addressed